anchore_engine/utils.py

```diff
@@ -213,11 +215,44 @@ def shellcheck(x):
         if not re.search("[;&<>]", x):
             return x
         else:
-            raise Exception("bad character in shell input")
+            raise Exception(SANITIZE_CMD_ERROR_MESSAGE)
 
     return [x for x in cmd_list if shellcheck(x)]
 
 
+def run_piped_command_list(
+    cmd_lists, stdout=subprocess.PIPE, stderr=subprocess.PIPE, **kwargs
+):
+    """
+    Executes a series of subcommands from cmd_lists, the stdout result of each being piped stdin for the next.
+    For example, the command: 'echo {}' | grype
+    cmd_lists should be passed as: [['echo', '{}'], ['grype']]
+    Do not include the actual pipe symbol, it is inferred from the data structure.
+    """
+    if not cmd_lists:
+        raise ValueError(PIPED_CMD_VALUE_ERROR_MESSAGE)
+    else:
+        output = None
+        for cmd_list in cmd_lists:
```

Comment:
This is a nice way of writing a readable for-loop. `for book in books` is the usual example I give, and this implements it well. Thanks for writing it this way